### PR TITLE
platforms/neuron: introduce NKI stack with SDPA kernel

### LINKED ISSUE #1339: Neuron PJRT StableHLO compile rewrites sort+slice sampling path to AwsNeuronTopK with full vocab size as k
### Describe the bug

We operate an inference stack that compiles model graphs directly to the AWS Neuron PJRT plugin from StablehloHLO and shardy MLIR dialects. The workload is not using PyTorch/XLA or JAX as the frontend. We generate StableHLO ourselves, pass it to `PJRT_Client_Compile` with program format `mlir`, and rely on libneuronxla to compile it.

On the Neuron 2.30 stack with libneuronxla 3.0, a sampling graph containing a descending `stablehlo.sort` over logits followed by a top-3 slice fails during compilation. The compiler appears to rewrite the sort/slice pattern into an internal Neuron top-k custom call, but the generated `AwsNeuronTopK` uses the full `vocab` dimension (`128256`) as `k` instead of the slice size (`3`).

The observed compiler diagnostic is:
```
[NCC_EVRF014] TopK custom operation does not support k values larger than 16384. Found k=128256.
%custom-call.72 = (bf16[64,128256]{1,0}, s32[64,128256]{1,0}) custom-call(%reshape.71), custom_call_target="AwsNeuronTopK", backend_config="128256"
```

The source StableHLO pattern asks for top 3:
```
%0 = stablehlo.iota dim = 0 : tensor<128256xi32>
%1 = stablehlo.broadcast_in_dim %0, dims = [1] : (tensor<128256xi32>) -> tensor<64x128256xi32>
%2:2 = "stablehlo.sort"(%arg0, %1) <{dimension = 1 : i64, is_stable = true}> ({
^bb0(%arg2: tensor<bf16>, %arg3: tensor<bf16>, %arg4: tensor<i32>, %arg5: tensor<i32>):
  %33 = stablehlo.compare GT, %arg2, %arg3, FLOAT : (tensor<bf16>, tensor<bf16>) -> tensor<i1>
  stablehlo.return %33 : tensor<i1>
}) : (tensor<64x128256xbf16>, tensor<64x128256xi32>) -> (tensor<64x128256xbf16>, tensor<64x128256xi32>)
%3 = stablehlo.slice %2#0 [0:64, 0:3] : (tensor<64x128256xbf16>) -> tensor<64x3xbf16>
%5 = stablehlo.slice %2#1 [0:64, 0:3] : (tensor<64x128256xi32>) -> tensor<64x3xi32>
```
`128256` is the vocab/sort dimension. The requested top-k size is represented by the slice limit `0:3`.

If the compiler rewrites the idiom to `AwsNeuronTopK`, use the slice size (3) as the top k, not the full sorted dimension.

### Model Name

Observed in LLM inference sampling for different models.. One concrete workload was a Llama 3 model. The issue is not model-weight-specific.

### Describe the workload type

We are running our own inference server/frontend on top of StableHLO and the Neuron PJRT plugin:

This is important because the issue is not currently a PyTorch/XLA or JAX issue. A JAX `lax.top_k` control lowers to `chlo.top_k`, and a JAX sort-based control emits a different float-sort comparator:
```
%neg = stablehlo.negate %arg0
%sorted:2 = "stablehlo.sort"(%neg, %indices) ...
  %cmp = stablehlo.compare LT, %canonical_lhs, %canonical_rhs, TOTALORDER
%top_neg = stablehlo.slice %sorted#0 [0:64, 0:3]
%top = stablehlo.negate %top_neg
```
Our failing frontend emits plain StableHLO sort/slice.

### Instance Type

inf2.8xlarge

### Release version

Neuron SDK stack: 2.30
neuronx-cc: 2.25.3371.0+f524f7f8
libneuronxla: 3.0.2891.0
aws-neuronx-runtime-lib: 2.32.31.0
aws-neuronx-collectives: 2.32.28.0

### Reproduction Steps

The source StableHLO sort/slice part is:
```
%sorted_values_and_indices:2 = "stablehlo.sort"(%logits, %indices)
  <{dimension = 1 : i64, is_stable = true}> ({
^bb0(%lhs: tensor<bf16>, %rhs: tensor<bf16>, %lhs_idx: tensor<i32>, %rhs_idx: tensor<i32>):
  %before = stablehlo.compare GT, %lhs, %rhs, FLOAT
  stablehlo.return %before : tensor<i1>
}) : (tensor<64x128256xbf16>, tensor<64x128256xi32>)
  -> (tensor<64x128256xbf16>, tensor<64x128256xi32>)

%top_values = stablehlo.slice %sorted_values_and_indices#0 [0:64, 0:3]
%top_indices = stablehlo.slice %sorted_values_and_indices#1 [0:64, 0:3]
```

### Regression Issue

- [ ] Select this option if this issue appears to be a regression.

### Possible Solution

If the sort/slice to topK rewrite is intended for this StableHLO pattern, the rewrite should use the post-sort slice size along the sorted dimension as k

### Logs/Context/Additional Information

_No response_

### PR DESCRIPTION
**Pull request by hugomano**
**Summary**

This PR introduces the Neuron NKI stack into ZML and wires it into the LLM attention path.

The main goal is to make Neuron attention run through NKI kernels.
It adds the compiler/runtime plumbing needed to embed NKI kernels as Neuron custom-native calls, then uses that path for SDPA-style prefill/decode attention. Paged Attention will come later.

On inf2.8xlarge it runs :
- 25 tok/s for Llama 3.1 8B BF16
- 63.5 tok/s for LFM2.5-1.2B-Instruct BF16
- 19 tok/s for Qwen3.5-9B BF16

**Neuron NKI compiler integration**

Adds a new `zml.ops.neuronNki(...)` API that lowers a Zig tensor operation to a StableHLO custom call with:

- `call_target_name = "AwsNeuronCustomNativeKernel"`
- generated operand/result layouts
- input/output tensor signatures derived from ZML shapes and dtypes
- a compiled NKI `backend_config` attached to the custom call

The compiler bridge lives in:

- `platforms/neuron/nki_kernel.zig`
- `platforms/neuron/nki_kernel_compiler.py`
- `platforms/neuron/nki_compiler_launcher.zig`
- `platforms/neuron/python_launcher.zig`

At graph construction time, ZML resolves the NKI source file from Bazel runfiles, writes a JSON compile request, invokes the sandboxed `nki-cc` launcher, and reads back the base64 backend config expected by Neuron’s custom-native kernel path.

The compiler target is inferred from the actual Neuron instance through `nrt_get_instance_info`, mapping supported families to targets.

**Neuron runtime**

The Neuron platform packaging is updated.

The previous setup carried a local `libneuronxla.zig` shim and initialized Python directly from the PJRT proxy so ZML could expose the compiler pieces expected by Neuron. With `libneuronxla` v0.3, the AWS package now owns more of that compiler/runtime surface, free of python, so the sandbox is rebuilt around the packaged `libneuronxla` payload instead of ZML’s local replacement.

- adds `nki>=0.4.0,<0.5`
- includes `aws-neuronx-tools`, `libgcc_s1`, and `libstdcpp6` required as needed by the so library
- exposes sandboxed `neuronx-cc` and `nki-cc` binaries
- moves Python initialization out of the PJRT shim and into reusable launcher code
- configures Neuron compiler/runtime env from the PJRT wrapper, including `NEURON_CC_FLAGS`, sandbox `PATH` etc.
- use Shardy instead of GSPMD

**NKI attention backend**

Adds a new attention backend:

- `zml.attention.attention.Backend.nki`
- selected automatically for `.neuron`
- exposed in `examples/llm --backend=nki`
- wired into Llama metadata/session handling

⚠️ We have an [issue](https://github.com/aws-neuron/aws-neuron-sdk/issues/1346) for Paged Attention, kernels will be added in another PR.

**StableHLO/lowering compatibility fixes for Neuron**

I added shims in a  few Neuron-sensitive lowering paths:
- keep scalar integer broadcasts as scalar broadcasts before StableHLO emission
- preserve gather fill semantics by replacing inactive sentinel lanes before backend indirect-memory lowering
- preserve scatter drop semantics by replacing inactive sentinel indices and masking inactive additive updates to zero
- keep downstream resharding after the scalar gather fast path, preventing Neuron SPMD from hoisting an all-gather before a narrow dynamic slice

These shims are isolated in `zml.ops.LoweringCompatibility` so model code can continue expressing normal StableHLO semantics. The gather/scatter sentinel handling is related to upstream Neuron issue https://github.com/aws-neuron/aws-neuron-sdk/issues/1335

Also for `topk` a specific branch has been added until the Neuron team fix https://github.com/aws-neuron/aws-neuron-sdk/issues/1339 . And I asked for a in graph level data movement API: https://github.com/aws-neuron/aws-neuron-sdk/issues/1340

**Neuron profiling workflow**

Adds a repo-level Neuron profiling workflow with documentation:

- `--config=neuron-profile`
- `//tools/neuron:profile`
- `//tools/neuron:server`
- `//tools/neuron:ingest`
- `//tools/neuron:summary-json`
- `//tools/neuron:summary-txt`
- `//tools/neuron:summary-perfetto`

**Known follow-ups**

- MoE NKI kernels
- Test kernels performance / adapt for trn2 and trn3

**Related doc**
- https://awsdocs-neuron.readthedocs-hosted.com/en/latest/nki/deep-dives/nki-compiler.html
- https://github.com/aws-neuron/nkipy/blob/main/nkipy/src/nkipy/core/backend/hlo.py

### CODE REVIEW

File: zml/BUILD.bazel [hunk 1]
```diff
@@ -69,6 +69,7 @@ zig_library(
     name = "zml",
     tags = ["manual"],
     data = [
+        "//zml/attention/nki:attention_kernel",
         "moe/triton_kernels/config.json",
```
Comment (Corendos) [inline zml/BUILD.bazel]: Shouldn't these be behind a `select` ?

**Review by Corendos — APPROVED** [review]

File: zml/BUILD.bazel (hunk 1 above)
Comment (hugomano) [reply zml/BUILD.bazel]: Done
